### ISSUE TITLE
Cursor remains hidden after closing the popup

### ISSUE THREAD

**Issue by isaif** [folke/noice.nvim#114]
**Describe the bug**
Cursor remain hidden after closing the popup. This doesn't happen if I use minimal setup provided.
How should I go about finding the cause of this issue?

**Which version of Neovim are you using?**
nightly

**To Reproduce**
1. open command line ":" or search "/"
2. run it or close it
3. cursor is not visible anymore

**Expected Behavior**
Cursor should be visible

![image](https://user-images.githubusercontent.com/11773308/197758787-b86c2c87-5719-4626-b36f-09ef1c2a9074.png)

![image](https://user-images.githubusercontent.com/11773308/197757504-9f59ec6c-6fee-4f35-8a99-bcfbe15ac6f3.png)

cursor is not visible anymore
![image](https://user-images.githubusercontent.com/11773308/197759043-455807a5-fea6-4a16-8d99-b187cc3dae42.png)

**Noice Log**
Please include any related errors from the Noice log file. (open with `:Noice log`)

<details>
<summary>Noice log</summary>
<pre>
...m/site/pack/packer/opt/noice.nvim/lua/noice/view/nui.lua:162: Failed to delete line

stack traceback:
	[C]: in function 'pcall'
	.../site/pack/packer/opt/noice.nvim/lua/noice/util/call.lua:140: in function <.../site/pack/packer/opt/noice.nvim/lua/noice/util/call.lua:139>
	[C]: in function 'nvim_buf_set_lines'
	...m/site/pack/packer/opt/noice.nvim/lua/noice/view/nui.lua:162: in function <...m/site/pack/packer/opt/noice.nvim/lua/noice/view/nui.lua:148>
	[C]: in function 'xpcall'
	.../site/pack/packer/opt/noice.nvim/lua/noice/util/call.lua:145: in function 'try'
	.../site/pack/packer/opt/noice.nvim/lua/noice/view/init.lua:122: in function 'display'
	...
	[C]: in function 'xpcall'
	.../site/pack/packer/opt/noice.nvim/lua/noice/util/call.lua:145: in function 'try'
	...site/pack/packer/opt/noice.nvim/lua/noice/util/hacks.lua:92: in function 'cmd'
	...ack/packer/start/nvim-notify/lua/notify/service/init.lua:63: in function 'push'
	...site/pack/packer/opt/noice.nvim/lua/noice/util/hacks.lua:200: in function 'push'
	...m/site/pack/packer/start/nvim-notify/lua/notify/init.lua:223: in function 'notify'
	.../site/pack/packer/opt/noice.nvim/lua/noice/util/init.lua:183: in function 'notify'
	.../site/pack/packer/opt/noice.nvim/lua/noice/util/init.lua:227: in function <.../site/pack/packer/opt/noice.nvim/lua/noice/util/init.lua:226>
	[C]: in function 'pcall'
	.../site/pack/packer/opt/noice.nvim/lua/noice/util/call.lua:124: in function <.../site/pack/packer/opt/noice.nvim/lua/noice/util/call.lua:122>

</pre>
</details>

**Comment by folke** [comment]
Does the cursor show when doing `:lua require("noice.util.hacks").show_cursor()` after that?

**Comment by isaif** [comment]
> Does the cursor show when doing `:lua require("noice.util.hacks").show_cursor()` after that?

No, it doesn't.

**Comment by folke** [comment]
Can you try again? Just pushed some changes

**Comment by isaif** [comment]
I did the packersync and still same problem exists.

`:lua require("noice.util.hacks").show_cursor()` doesn't help either.

**Comment by folke** [comment]
That's really odd. Do you set `guicursor` somewhere in your config?

What does `:set guicursor?` give?

And does `:set guicursor=` restore the cursor?

Are you using a gui?

**Comment by isaif** [comment]
I had ` vim.opt.guicursor = '' "` to keep cursor as block in insert mode. I didn't know this caused the issue.

Sorry for taking so much of your time.

If I may ask if there is another way to keep cursor as block in insert mode.

Thanks for your time.

**Comment by folke** [comment]
It's weird that that worked before. Setting the `guicursor=""`, disables the cursor I believe.

Check for the options `:h guicursor`

**Comment by gongfarmer** [comment]
+1.  I hit this same problem.
Had to stop using noice for a few days, until I found this bug report.
If this is common enough, might be worth a warning in checkhealth.

**Comment by folke** [comment]
Something weird is happening with `guicursor=""`. I'm looking into it.

**Comment by folke** [comment]
I think I was able to fix it. Something weird is happening internally with Neovim and setting the guicursor. It seems that setting it, only comes into effect after some time. And setting it again in between it is active, doesn't use the new value.

Adding a delay before resetting it seems to fix it

**Comment by isaif** [comment]
I changed it to `vim.opt.guicursor = 'a:block'` instead and the issue was fixed.

**Comment by theherk** [comment]
I'm in a similar situation, and though this is closed I'm still not clear on the resolution. I set `vim.o.guicursor = vim.o.guicursor .. ",a:blinkon1"`, but when closing a popup from https://github.com/voldikss/vim-browser-search, my cursor is gone. Neither resetting the value of `guicursor` nor using the `show_cursor()` function has any effect. The value remains `a:NoiceHiddenCursor`. What have I overlooked?
